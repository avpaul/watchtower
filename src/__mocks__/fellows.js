export default [
  {
    id: 1,
    firstName: 'Mathilda',
    lastName: 'West',
    level: 'D0A Simulations',
    quantity: 1,
    quality: 0,
    initiative: 0.3,
    communication: 0,
    professionalism: 0,
    integration: 0,
  },
  {
    id: 2,
    firstName: 'Mittie',
    lastName: 'Bowers',
    level: 'D0B Apprenticeships',
    quantity: 1,
    quality: 0,
    initiative: 0.4,
    communication: 0.1,
    professionalism: 0,
    integration: 0,
  },
  {
    id: 3,
    firstName: 'Adekunle',
    lastName: 'Gold',
    level: 'D0B Apprenticeships',
    quantity: 0,
    quality: 1,
    initiative: 0.8,
    communication: 0,
    professionalism: 0,
    integration: 1,
  },
  {
    id: 4,
    firstName: 'Hallie',
    lastName: 'James',
    level: 'D0A Simulations',
    quantity: 0,
    quality: 0,
    initiative: 1,
    communication: 0.4,
    professionalism: 0,
    integration: 0,
  },
  {
    id: 5,
    firstName: 'Estella',
    lastName: 'Graves',
    level: 'D0A Simulations',
    quantity: 1,
    quality: 0,
    initiative: 1,
    communication: 0.9,
    professionalism: 0,
    integration: 1,
  },
  {
    id: 6,
    firstName: 'Genevieve',
    lastName: 'Black',
    level: 'D0B Apprenticeships',
    quantity: 1,
    quality: 1,
    initiative: 0.3,
    communication: 0.4,
    professionalism: 1,
    integration: 0,
  },
  {
    id: 7,
    firstName: 'Gavin',
    lastName: 'Berry',
    level: 'D0A Simulations',
    quantity: 0.4,
    quality: 0,
    initiative: 0.4,
    communication: 1,
    professionalism: 0,
    integration: 1,
  },
  {
    id: 8,
    firstName: 'Tonny',
    lastName: 'Gibbs',
    level: 'D0B Apprenticeships',
    quantity: 0,
    quality: 1,
    initiative: 1,
    communication: 0.3,
    professionalism: 1,
    integration: 1,
  },
  {
    id: 9,
    firstName: 'Chase',
    lastName: 'Wright',
    level: 'D0B Apprenticeships',
    quantity: 0.1,
    quality: 0.4,
    initiative: 0.4,
    communication: 1,
    professionalism: 0,
    integration: 1,
  },
  {
    id: 10,
    firstName: 'Hallie',
    lastName: 'Sullivan',
    level: 'D0A Simulations',
    quantity: 0.2,
    quality: 0.1,
    initiative: 1,
    communication: 0,
    professionalism: 0,
    integration: 1,
  },
];
