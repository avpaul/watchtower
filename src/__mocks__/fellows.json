[
  {
    "email": "abbey.maddy@andela.com",
    "level": "D0A Simulations",
    "cohort": "Class 1 - NBO",
    "actual apprenticeship start date": "2018-12-10",
    "simulation start date": "2018-11-01",
    "fellow_id": "-LMKN5flPv4R7LCZs0lh",
    "staffId": null,
    "firstName": "Abbey",
    "lastName": "Maddy",
    "submitterFirstName": "",
    "submitterLastName": "",
    "status": "onTrack",
    "isBeforeWk5": true,
    "quality": "1.00",
    "integration": "1.00",
    "professionalism": "1.00",
    "communication": "1.00",
    "initiative": "1.00",
    "quantity": "1.00",
    "devPulseStatus": "On Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": 23
  },
  {
    "fellow_id": "-LJK1qECFdekucrwqqkh",
    "staffId": "-LAmk78f7GyBWkt2UWpb",
    "submitterFirstName": "Ephraim",
    "submitterLastName": "Malinga",
    "email": "gordon.balaaka@andela.com",
    "firstName": "Gordon",
    "lastName": "Balaaka",
    "level": "D0A Simulations",
    "cohort": "Class 10 - KLA",
    "isBeforeWk5": false,
    "status": "gteWk5OffTrack",
    "quality": "1.00",
    "quantity": "1.12",
    "initiative": "1.00",
    "communication": "1.12",
    "professionalism": "1.00",
    "integration": "1.25",
    "levelReturned": "D0A Simulations",
    "devPulseStatus": "Off Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": "N/A"
  },
  {
    "email": "timothy.mubiru@andela.com",
    "level": "D0A Simulations",
    "cohort": "Class 13 - KLA",
    "actual apprenticeship start date": "2018-10-15",
    "expected apprenticeship completion date": "2019-01-16",
    "fellow_id": "-LJK2KDhq70_X1YbirFx",
    "staffId": "-LAmk78f7GyBWkt2UWpb",
    "firstName": "Timothy",
    "lastName": "Mubiru",
    "submitterFirstName": "Ephraim",
    "submitterLastName": "Malinga",
    "status": "onTrack",
    "isBeforeWk5": true,
    "quality": "1.00",
    "integration": "1.00",
    "professionalism": "1.00",
    "communication": "1.00",
    "initiative": "1.00",
    "quantity": "1.00",
    "devPulseStatus": "On Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": "N/A"
  },
  {
    "email": "sam.rubarema@andela.com",
    "level": "D0B Apprenticeship",
    "cohort": "Class 12 - KLA",
    "actual apprenticeship start date": "2019-02-18",
    "actual simulations completion date": "2019-02-14",
    "simulation start date": "2018-11-19",
    "expected apprenticeship completion date": "2019-05-16",
    "fellow_id": "-LP641ReDxZeT2ABACW8",
    "staffId": null,
    "firstName": "Sam",
    "lastName": "Rubarema",
    "submitterFirstName": "",
    "submitterLastName": "",
    "status": "onTrack",
    "isBeforeWk5": true,
    "quality": "1.00",
    "integration": "1.00",
    "professionalism": "1.00",
    "communication": "1.00",
    "initiative": "1.00",
    "quantity": "1.00",
    "devPulseStatus": "On Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": 7
  },
  {
    "email": "fredrick.adegoke@andela.com",
    "level": "D0B Apprenticeship",
    "cohort": "Class 41 - LOS",
    "actual apprenticeship start date": "2019-01-14",
    "actual simulations completion date": "2019-01-11",
    "simulation start date": "2018-10-16",
    "expected apprenticeship completion date": "2019-04-08",
    "fellow_id": "-LLUW2BwQiZS0j9KF10j",
    "staffId": "-LGy4OuPDHCZCZvDuPz0",
    "firstName": "Fredrick",
    "lastName": "Adegoke",
    "submitterFirstName": "Trust",
    "submitterLastName": "Birungi",
    "status": "onTrack",
    "isBeforeWk5": true,
    "quality": "1.00",
    "integration": "1.00",
    "professionalism": "1.00",
    "communication": "1.00",
    "initiative": "1.00",
    "quantity": "1.00",
    "devPulseStatus": "On Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": 12
  },
  {
    "fellow_id": "-LEnW5RRn0IkO8VjnMma",
    "staffId": "-LGy4OuPDHCZCZvDuPz0",
    "submitterFirstName": "Trust",
    "submitterLastName": "Birungi",
    "email": "sinmiloluwa.oloyede@andela.com",
    "firstName": "Sinmiloluwa",
    "lastName": "Oloyede",
    "level": "D0B Apprenticeship",
    "cohort": "Class 38 - LOS",
    "isBeforeWk5": false,
    "status": "gteWk5OffTrack",
    "quality": "1.10",
    "quantity": "1.30",
    "initiative": "1.20",
    "communication": "1.00",
    "professionalism": "1.10",
    "integration": "1.10",
    "levelReturned": "D0B Apprenticeship",
    "devPulseStatus": "Off Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": "N/A"
  },
  {
    "fellow_id": "-LJK2fj_k_R7aYBAq_AR",
    "staffId": "-LJXG90epIwmW2U1XQHB",
    "submitterFirstName": "Martin",
    "submitterLastName": "Bbaale",
    "email": "samuel.bwambale@andela.com",
    "firstName": "Samuel",
    "lastName": "Bwambale",
    "level": "D0A Simulations",
    "cohort": "Class 10 - KLA",
    "isBeforeWk5": false,
    "status": "gteWk5OffTrack",
    "quality": "1.00",
    "quantity": "1.12",
    "initiative": "1.25",
    "communication": "1.25",
    "professionalism": "1.50",
    "integration": "1.50",
    "levelReturned": "D0A Simulations",
    "devPulseStatus": "Off Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": "N/A"
  },
  {
    "fellow_id": "-LD3PA3VgP4qvRRORX08",
    "staffId": "-LGy4OuPDHCZCZvDuPz0",
    "submitterFirstName": "Trust",
    "submitterLastName": "Birungi",
    "email": "folajimi.ogunbadejo@andela.com",
    "firstName": "Folajimi",
    "lastName": "Ogunbadejo",
    "level": "D0B Apprenticeship",
    "cohort": "Class 37 - LOS",
    "isBeforeWk5": false,
    "status": "gteWk5OffTrack",
    "quality": "1.00",
    "quantity": "1.41",
    "initiative": "1.33",
    "communication": "0.00",
    "professionalism": "0.00",
    "integration": "1.00",
    "levelReturned": "D0B Apprenticeship",
    "devPulseStatus": "Off Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": "N/A"
  },
  {
    "fellow_id": "-LJK2KDhq70_X1YbirFa",
    "staffId": "-LAmk78f7GyBWkt2UWpb",
    "submitterFirstName": "Ephraim",
    "submitterLastName": "Malinga",
    "email": "edward.kigozi@andela.com",
    "firstName": "Edward",
    "lastName": "Kigozi",
    "level": "D0A Simulations",
    "cohort": "Class 10 - KLA",
    "isBeforeWk5": true,
    "status": "ltWk5OffTrack",
    "quality": "1.33",
    "quantity": "1.33",
    "initiative": "1.33",
    "communication": "1.33",
    "professionalism": "1.33",
    "integration": "1.33",
    "levelReturned": "D0A Simulations",
    "devPulseStatus": "Off Track",
    "lmsStatus": "N/A",
    "advanceStatus": "N/A",
    "weeksSpent": "N/A"
  }
]
