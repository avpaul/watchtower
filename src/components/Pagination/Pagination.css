#pagination-container {
  display: flex;
  justify-content: center;
}

#pagination {
  display: flex;
  color: #505050;
}

.first {
  height: 36px;
  width: 55px;
  border: 0.5px solid rgba(0, 0, 0, 0.2);
  border-radius: 2px;
  background-color: #ffffff;
  box-shadow: 0 1px 1px 0 rgba(221, 221, 221, 0.5);
  color: #000000;
  font-family: 'DIN Pro';
}

.second {
  height: 36px;
  width: 55px;
  border: 0.5px solid rgba(0, 0, 0, 0.2);
  border-radius: 2px;
  background-color: #ffffff;
  box-shadow: 0 1px 1px 0 rgba(221, 221, 221, 0.5);
  color: #3359db;
  font-family: 'DIN Pro';
}

.per-page-label {
  padding: 0px;
}

.showing {
  height: 20px;
  width: 145px;
  color: #637381;
  font-family: 'DIN Pro';
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

.per-page {
  padding: 0px 20px 5px 10px !important;
}

.select {
  padding: 0px 20px 5px 10px !important;
}
