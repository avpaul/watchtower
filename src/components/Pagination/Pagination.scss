@import '../../assets/sass/abstract/variables';

#pagination-container {
  display: flex;
  justify-content: center;
}

#pagination {
  display: flex;
  color: #505050;
}

.pg {
  position: relative;
}

.pg__per-page {
  width: 4rem;
}

.pg__buttons .btn:nth-child(1),
.pg__buttons .btn:nth-child(7) {
  margin: 0 1rem;
}

.pg__button,
.pg__button--active {
  height: 36px;
  width: 55px;
  border: 0.5px solid rgba(0, 0, 0, 0.2);
  border-radius: 2px;
  background-color: #ffffff;
  box-shadow: 0 1px 1px 0 rgba(221, 221, 221, 0.5);
  color: #000000;
  font-family: $font-primary;
}

.pg__button {
  background-color: #ffffff;
  color: #000000;
}

.pg__button--active {
  background-color: #3359db;
  color: #ffffff;
}

.pg__showing {
  height: 20px;
  width: 145px;
  color: #637381;
  font-family: $font-primary;
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

.pg__per-page__label {
  padding-right: 20px;
  padding-left: 10px;
  line-height: 30px;
}

.pg__per-page__select {
  padding-right: 20px;
  padding-left: 10px;
}
