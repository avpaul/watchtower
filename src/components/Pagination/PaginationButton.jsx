import React from 'react';
import PropTypes from 'prop-types';

const PaginationButton = ({
  className,
  name,
  value,
  onClick,
  buttonText,
  disableBool = false
}) => (
  <button
    type="button"
    className={className}
    value={value}
    name={name}
    onClick={onClick}
    disabled={disableBool}
    key={value}
  >
    {buttonText}
  </button>
);

PaginationButton.propTypes = {
  className: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  onClick: PropTypes.func.isRequired,
  buttonText: PropTypes.string.isRequired,
  disableBool: PropTypes.bool
};

PaginationButton.defaultProps = {
  disableBool: false
};

export default PaginationButton;
