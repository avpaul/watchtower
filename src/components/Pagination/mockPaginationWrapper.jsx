const mockPaginationWrapper = {
  state: {
    paginationFilter: {
      perPage: 25,
      page: 1,
      totalPages: 0
    },
    paginatedData: [],
    filteredData: []
  },
  updateData: () => {},
  paginateData: () => {},
  processData: () => {},
  handlePaginationPageChange: () => {},
  renderPagination: () => {}
};

export default mockPaginationWrapper;
