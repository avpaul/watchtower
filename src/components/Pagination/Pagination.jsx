import React, { Component } from 'react';
import './Pagination.css';
import PropTypes from 'prop-types';

/**
 * class representing the pagination component
 * @class
 */

class Pagination extends Component {
  /**
   * changes page based on next or previous click
   * @param { object } - event that happens
   */
  onPageChange = event => {
    const { handlePageChange, perPage, filter } = this.props;
    handlePageChange(
      `${event.target.value}&perPage=${perPage}&filter=${filter}`
    );
  };

  /**
   * based on button click
   * @param { object } - event that happens
   */
  onValueChange = event => {
    const { handleValueChange } = this.props;
    handleValueChange(event.target.value);
  };

  renderItems = number => {
    const disabledButton = false;
    const { currentPage } = this.props;
    const elements = [];
    let i = 0;
    while (i < number) {
      elements.push(i + 1);
      i += 1;
    }
    return (
      <div id="form-group col-md-4 row">
        {elements.map(element =>
          this.renderPaginationButtons(
            element === currentPage
              ? 'btn btn-default second active-page'
              : 'btn btn-default second',
            'previous',
            element,
            this.onValueChange,
            element,
            disabledButton
          )
        )}
      </div>
    );
  };

  renderNormal(
    { currentPage, nextPage, prevPageUrl, totalPages } = this.props
  ) {
    const disableObject = {
      disabled: false,
      disable: true
    };
    const buttonOneEllipsis = currentPage;
    const buttonTwoEllipsis = currentPage + 1;
    const buttonOneAfterEllipse = totalPages - 1;
    const buttonTwoAfterEllipse = totalPages;

    if (currentPage > 1) {
      disableObject.disable = false;
    }
    if (currentPage === totalPages) {
      disableObject.disabled = true;
    }
    return (
      <div id="form-group col-md-4 row">
        {this.renderPaginationButtons(
          'btn btn-default first',
          'first',
          prevPageUrl,
          this.onPageChange,
          'Prev',
          disableObject.disable
        )}
        <div
          className="btn-group mr-2 ml-2"
          role="group"
          aria-label="First group"
        >
          {this.renderPaginationButtons(
            currentPage
              ? 'btn btn-default first active-page'
              : 'btn btn-default first',
            'previous',
            currentPage,
            this.onValueChange,
            buttonOneEllipsis
          )}
          {this.renderPaginationButtons(
            'btn btn-default second',
            'previous',
            currentPage + 1,
            this.onValueChange,
            buttonTwoEllipsis
          )}
          <button type="button" className="btn btn-default second" value="">
            ...
          </button>
          {this.renderPaginationButtons(
            'btn btn-default second',
            'next',
            buttonOneAfterEllipse,
            this.onValueChange,
            buttonOneAfterEllipse
          )}
          {this.renderPaginationButtons(
            'btn btn-default second',
            'next',
            buttonTwoAfterEllipse,
            this.onValueChange,
            buttonTwoAfterEllipse
          )}
        </div>
        {this.renderPaginationButtons(
          'btn btn-default first',
          'next',
          nextPage,
          this.onPageChange,
          'Next',
          disableObject.disabled
        )}
      </div>
    );
  }

  renderPaginationButtons = (
    className,
    name,
    value,
    onClick,
    buttonText,
    disableBool = false
  ) => (
    <button
      type="button"
      className={className}
      value={value}
      name={name}
      onClick={onClick}
      disabled={disableBool}
      key={value}
    >
      {buttonText}
    </button>
  );

  renderButton = () => {
    const { totalPages } = this.props;
    if (totalPages < 5) {
      return this.renderItems(totalPages);
    }
    return this.renderNormal();
  };

  render() {
    return <div>{this.renderButton()}</div>;
  }
}

export default Pagination;

Pagination.propTypes = {
  totalPages: PropTypes.number,
  handlePageChange: PropTypes.func.isRequired,
  perPage: PropTypes.string,
  handleValueChange: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
  currentPage: PropTypes.number.isRequired
};

Pagination.defaultProps = {
  totalPages: 10,
  perPage: '25'
};
