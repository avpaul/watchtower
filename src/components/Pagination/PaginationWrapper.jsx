import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Pagination from './Pagination';

export default class PaginationFrontendWrapper extends Component {
  constructor(props) {
    super(props);

    this.state = {
      paginationFilter: {
        perPage: 25,
        page: 1,
        totalPages: 0
      },
      // eslint-disable-next-line react/no-unused-state
      paginatedData: [],
      filteredData: []
    };
  }

  updateData = (data, filters = {}) => {
    this.setState(
      state => ({
        filteredData: data,
        paginationFilter: {
          ...state.paginationFilter,
          ...filters,
          totalPages: Math.ceil(data.length / state.paginationFilter.perPage)
        }
      }),
      this.paginateData
    );
    return true;
  };

  paginateData = () => {
    const { paginationFilter, filteredData } = this.state;
    this.setState({
      // eslint-disable-next-line react/no-unused-state
      paginatedData: this.processData(filteredData, paginationFilter)
    });
  };

  /**
   * @description Filters the data records according to the current page
   * @param pagination Object that contains the values used to filter the data records eg.
   * current page, per page value
   */
  processData = (data, filters) => {
    const offset = (filters.page - 1) * filters.perPage;
    const upperBound = offset + filters.perPage;
    // Retrieves a sub array of the data according to current page and the page size set.
    const updatedData = data;
    return updatedData.slice(offset, upperBound);
  };

  /**
   * @description pagination buttons' onClick event handler
   * @param queryData pagination filters returned from the Pagination component
   */
  handlePaginationPageChange = queryData => {
    this.setState(
      state => ({
        paginationFilter: {
          perPage: queryData.perPage,
          page: queryData.page,
          totalPages: Math.ceil(state.filteredData.length / queryData.perPage)
        }
      }),
      this.paginateData
    );
  };

  renderPagination = () => {
    const { paginationFilter, filteredData } = this.state;
    return (
      <Pagination
        totalPages={paginationFilter.totalPages}
        handlePageChange={this.handlePaginationPageChange}
        handleValueChange={this.handlePaginationPageChange}
        currentPage={paginationFilter.page}
        perPage={paginationFilter.perPage}
        hasData={filteredData.length > 0}
      />
    );
  };

  render() {
    const { component } = this.props;
    return React.cloneElement(component, { paginationWrapper: this });
  }
}

PaginationFrontendWrapper.propTypes = {
  component: PropTypes.element.isRequired
};
