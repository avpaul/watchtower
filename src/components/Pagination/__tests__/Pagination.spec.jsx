import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../Pagination';
import fellows from '../../../__mocks__/fellows';

describe('Tests Pagination component', () => {
  let wrapper;
  let wrapper2;

  const props = {
    handlePageChange: jest.fn(),
    totalPages: 10,
    currentPage: 1,
    perPage: 10,
    prevPageUrl: '',
    nextPage: '2',
    filter: {
      perPage: 10,
      page: 1
    },
    handleValueChange: jest.fn(),
    onPerPageChange: jest.fn(),
    hasFellows: true
  };

  const testPaginationComponentMethod = (method, testWrapper = wrapper) => {
    const shallowWrapper = testWrapper.instance();
    const spy = jest.spyOn(shallowWrapper, method);
    shallowWrapper[method]();
    expect(spy).toHaveBeenCalledTimes(1);
  };

  beforeEach(() => {
    wrapper = shallow(<Pagination {...props} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls render normal', () => {
    testPaginationComponentMethod('renderNormal');
  });

  it('onPageChange should handle change on next ', () => {
    const event = {
      target: {
        name: 'next',
        value: '1'
      }
    };
    wrapper.instance().onPageChange(event);
  });

  it('calls render normal', () => {
    const newProps = {
      ...props,
      nextPage: '5',
      currentPage: 3
    };

    wrapper = shallow(<Pagination {...newProps} />);

    testPaginationComponentMethod('renderNormal', wrapper);
  });

  it('calls render total pages equals current page normal', () => {
    const newProps = {
      ...props,
      nextPage: '5',
      currentPage: 10
    };

    wrapper = shallow(<Pagination {...newProps} />);
    testPaginationComponentMethod('renderNormal', wrapper);
  });

  const testPageChange = (change, result) => {
    const newProps = {
      ...props,
      nextPage: '5',
      currentPage: 10
    };

    wrapper = shallow(<Pagination {...newProps} />);

    const event = {
      target: {
        name: change,
        value: '1'
      }
    };

    wrapper.instance().onPageChange(event);

    setTimeout(() => {
      expect(wrapper.state().page).toBe(result);
    }, 500);
  };

  it('onPageChange should handle change on first', () => {
    testPageChange('first', 11);
  });

  it('onPageChange should handle change on first', () => {
    testPageChange('previous', 9);
  });

  it('onPageChange should handle change on first', () => {
    testPageChange('next', 11);
  });

  it('onValueChange is called', () => {
    const shallowWrapper = wrapper.instance();
    const event = {
      target: {
        name: 'first',
        value: '1'
      }
    };
    shallowWrapper.onValueChange(event);
  });

  it('onPageChange should handle change on first', () => {
    const event = {
      target: {
        name: 'first',
        value: '1'
      }
    };
    wrapper.instance().onPageChange(event);
  });

  it('onPerPageChange should change perPage state', () => {
    const event = {
      target: {
        name: 'first',
        value: '50'
      }
    };
    wrapper.instance().onPerPageChange(event);

    setTimeout(() => {
      expect(wrapper.state().perPage).toBe(50);
    }, 500);
  });

  it('onValueChange is called', () => {
    wrapper.setState({ fellows });
    const button = wrapper.find('.pg__button').first();
    button.simulate('click', { target: { value: '1' } });
  });

  it('onPageChange should set state', () => {
    wrapper2 = shallow(<Pagination {...props} />);
    const event = {
      target: {
        name: 'first',
        value: '1'
      }
    };
    wrapper2.instance().onPageChange(event);
    wrapper2
      .instance()
      .setState({ disabled: 'true', count: '10', totalPages: '10' });
  });

  it('renders pagination button with totalpages as 3', () => {
    const newProps = {
      ...props,
      totalPages: 3,
      perPage: 1
    };

    wrapper = shallow(<Pagination {...newProps} />);
    wrapper.instance().componentDidUpdate({ ...newProps, perPage: 10 });
    expect(wrapper.state('perPage')).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
