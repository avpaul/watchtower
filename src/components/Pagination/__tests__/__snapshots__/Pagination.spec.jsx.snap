// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Pagination component matches snapshot 1`] = `
<div
  className="pg align-self-center"
>
  <div
    className="row d-flex justify-content-center"
  >
    <div
      className="align-self-center col-md-9 col-lg-7"
    >
      <p
        className="text-center"
      >
        Showing 
        1
         of 
        10
         pages
      </p>
      <div
        className="pg__filters row justify-content-center"
      >
        <div
          className="pg__per-page col-md-4 col-6 mt-4 mt-md-0 mb-4 mb-md-0 col-md-4"
        >
          <div
            className="row d-flex justify-content-center"
          >
            <div
              className="pg__per-page__select col-lg-4 col-6"
            >
              <select
                className="form-control d-flex justify-content-center pg__per-page"
                onChange={[Function]}
                value={10}
              >
                <option
                  key="25"
                  value={25}
                >
                  25
                </option>
                <option
                  key="50"
                  value={50}
                >
                  50
                </option>
                <option
                  key="100"
                  value={100}
                >
                  100
                </option>
              </select>
            </div>
            <div
              className="pg__per-page__label col-6"
            >
              Per page
            </div>
          </div>
        </div>
        <div
          className="col-md-8 col-12"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="form-group pg__buttons col"
            >
              <PaginationButton
                buttonText="Prev"
                className="btn btn-default pg__button"
                disableBool={false}
                disabledButton={false}
                name="previous"
                onClick={[Function]}
                value="Prev"
              />
              <PaginationButton
                buttonText="1"
                className="btn btn-default pg__button--active"
                disableBool={false}
                name="previous"
                onClick={[Function]}
                value={1}
              />
              <PaginationButton
                buttonText="2"
                className="btn btn-default pg__button"
                disableBool={false}
                name="previous"
                onClick={[Function]}
                value={2}
              />
              <button
                className="btn btn-default pg__button"
                type="button"
                value=""
              >
                ...
              </button>
              <PaginationButton
                buttonText="9"
                className="btn btn-default pg__button"
                disableBool={false}
                name="next"
                onClick={[Function]}
                value={9}
              />
              <PaginationButton
                buttonText="10"
                className="btn btn-default pg__button"
                disableBool={false}
                name="next"
                onClick={[Function]}
                value={10}
              />
              <PaginationButton
                buttonText="Next"
                className="btn btn-default pg__button"
                disableBool={false}
                disabledButton={true}
                name="next"
                onClick={[Function]}
                value="Next"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests Pagination component renders pagination button to match snapshots 1`] = `
<button
  className="btn btn-default pg__button"
  disabled={false}
  key="prev"
  name="previous"
  onClick={[Function]}
  type="button"
  value="prev"
>
  prev
</button>
`;

exports[`Tests Pagination component renders pagination button with totalpages as 3 1`] = `
<div
  className="pg align-self-center"
>
  <div
    className="row d-flex justify-content-center"
  >
    <div
      className="align-self-center col-md-6 col-lg-5"
    >
      <p
        className="text-center"
      >
        Showing 
        1
         of 
        3
         pages
      </p>
      <div
        className="pg__filters row justify-content-center"
      >
        <div
          className="pg__per-page col-md-4 col-6 mt-4 mt-md-0 mb-4 mb-md-0 col-md-5"
        >
          <div
            className="row d-flex justify-content-center"
          >
            <div
              className="pg__per-page__select col-lg-4 col-6"
            >
              <select
                className="form-control d-flex justify-content-center pg__per-page"
                onChange={[Function]}
                value={1}
              >
                <option
                  key="25"
                  value={25}
                >
                  25
                </option>
                <option
                  key="50"
                  value={50}
                >
                  50
                </option>
                <option
                  key="100"
                  value={100}
                >
                  100
                </option>
              </select>
            </div>
            <div
              className="pg__per-page__label col-6"
            >
              Per page
            </div>
          </div>
        </div>
        <div
          className="col-md-7 col-12"
        >
          <div
            className="col align-self-center text-center"
            id="form-group"
          >
            <PaginationButton
              buttonText="1"
              className="btn btn-default pg__button--active"
              disableBool={false}
              disabledButton={false}
              key="1"
              name="previous"
              onClick={[Function]}
              value={1}
            />
            <PaginationButton
              buttonText="2"
              className="btn btn-default pg__button"
              disableBool={false}
              disabledButton={false}
              key="2"
              name="previous"
              onClick={[Function]}
              value={2}
            />
            <PaginationButton
              buttonText="3"
              className="btn btn-default pg__button"
              disableBool={false}
              disabledButton={false}
              key="3"
              name="previous"
              onClick={[Function]}
              value={3}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
